x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/snapshot_list/snapshot_table/snapshot_table.tsx

```diff
@@ -102,34 +143,85 @@ export const SnapshotTable: React.FunctionComponent<Props> = ({
       },
     },
     {
-      field: 'indices',
-      name: i18n.translate('xpack.snapshotRestore.snapshotList.table.indicesColumnTitle', {
-        defaultMessage: 'Indices',
-      }),
-      truncateText: true,
-      sortable: true,
-      width: '100px',
-      render: (indices: string[]) => indices.length,
-    },
-    {
-      field: 'shards.total',
-      name: i18n.translate('xpack.snapshotRestore.snapshotList.table.shardsColumnTitle', {
-        defaultMessage: 'Shards',
-      }),
-      truncateText: true,
-      sortable: true,
-      width: '100px',
-      render: (totalShards: number) => totalShards,
-    },
-    {
-      field: 'shards.failed',
-      name: i18n.translate('xpack.snapshotRestore.snapshotList.table.failedShardsColumnTitle', {
-        defaultMessage: 'Failed shards',
+      name: i18n.translate('xpack.snapshotRestore.snapshotList.table.actionsColumnTitle', {
+        defaultMessage: 'Actions',
       }),
-      truncateText: true,
-      sortable: true,
+      actions: [
+        {
+          render: ({ snapshot, repository, state }: SnapshotDetails) => {
+            const canRestore = state === SNAPSHOT_STATE.SUCCESS || state === SNAPSHOT_STATE.PARTIAL;
+            const label = canRestore
+              ? i18n.translate('xpack.snapshotRestore.snapshotList.table.actionRestoreTooltip', {
+                  defaultMessage: 'Restore',
+                })
+              : state === SNAPSHOT_STATE.IN_PROGRESS
+              ? i18n.translate(
+                  'xpack.snapshotRestore.snapshotList.table.actionRestoreDisabledInProgressTooltip',
+                  {
+                    defaultMessage: `Can't restore in-progress snapshot`,
+                  }
+                )
+              : i18n.translate(
+                  'xpack.snapshotRestore.snapshotList.table.actionRestoreDisabledInvalidTooltip',
+                  {
+                    defaultMessage: `Can't restore invalid snapshot`,
+                  }
+                );
+            return (
+              <EuiToolTip content={label}>
+                <EuiButtonIcon
+                  aria-label={i18n.translate(
+                    'xpack.snapshotRestore.snapshotList.table.actionRestoreAriaLabel',
+                    {
+                      defaultMessage: 'Store snapshot `{name}`',
+                      values: { name: snapshot },
+                    }
+                  )}
+                  iconType="exportAction"
+                  color="primary"
+                  data-test-subj="srsnapshotListRestoreActionButton"
+                  href={`#${BASE_PATH}/restore_snapshot/${repository}/${snapshot}`}
```

Comment:
I haven't tested this but it looks like `BASE_PATH` already contains `snapshot_restore/` so is this link broken?

We can fix and future-proof this by generating this URL via a function in the `services/navigation/links.ts` file and calling it from here. Same goes for other URLs in this PR.

Reply:
this path is `restore_snapshot/`, not `snapshot_restore/` :D however this is very confusing, so I will change this path so that the url is not `/snapshot_restore/restore_snapshot` lol

Reply:
And this is the story of how I learned I'm dyslexic!